### PR TITLE
Spec Parser: Move generated spec parser to package

### PR DESCRIPTION
**Pull request by aduth**
This pull request seeks to treat `lib/parser.php` as a generated build artifact of the `block-serialization-spec-parser` package, both removing it from `lib/` and from source altogether, assumed to be generated as part of `npm run build` in time for a package publish. This treatment has been applied to the `parser.js` generated result as well.

**Testing instructions:**

`lib/parser.php` is not actually used by Gutenberg, so it should have no impact on the effective behavior of the plugin.

Verify that `parser.js` and `parser.php` are generated to `packages/block-serialization-spec-parser` after `npm install`.

Ensure that tests pass covering the generated files:

```
npm run test-unit packages/block-serialization-spec-parser
```

### CODE REVIEW

File: bin/run-wp-unit-tests.sh [hunk 1]
```diff
@@ -21,16 +21,6 @@ fi
 
 npm run build || exit 1
 
-# Make sure phpegjs parser is up to date
-node bin/create-php-parser.js || exit 1
-if ! git diff --quiet --exit-code lib/parser.php; then
```
Comment (aduth) [inline bin/run-wp-unit-tests.sh]: This check is already accounted for in CI by:

- The files are automatically built as a `postinstall` step
- CI runs [`check-local-changes`](https://github.com/WordPress/gutenberg/blob/ab0acc399ea3457833b2f7f50fd9a3bff90a57d0/package.json#L155) to verify no commit-able files

**Review by dmsnell — APPROVED** [review]
Thanks @aduth! This seems like a good improvement on code organization and reducing noise in the repo.

File: bin/run-wp-unit-tests.sh (hunk 1 above)
Comment (gziolo) [reply bin/run-wp-unit-tests.sh]: 👍

**Review by gziolo — APPROVED** [review]
Yes, this is all good. As long as those files land on npm they can be ingored from git repository.

Very nice refactor.

**Comment by aduth** [comment]
>As long as those files land on npm they can be ingored from git repository.

I think this might not actually be working as expected yet. I learned recently that apparently npm will inherit a `gitignore` when there's no other whitelist / blacklist. I tracked it down to:

>If there is a .gitignore file, and .npmignore is missing, .gitignore’s contents will be used instead.

https://docs.npmjs.com/files/package.json#files

Indeed, running `npm pack && tar -tf wordpress-block-serialization-spec-parser-2.0.3.tgz` revealsa there is no `parser.js` nor `parser.php`.

We could define `files`, or alternatively just add an empty `.npmignore`. I'll see which works better.

**Comment by aduth** [comment]
Started with a whitelist, but apparently some of the main exports of the package include tests, so switched to an empty `.npmignore` instead, which appears to work well.

**Comment by gziolo** [comment]
Thanks for exploring it further. I took it by surprise that it works this way. It should be okey to proceed as is 💯

**Comment by mcsf** [comment]
Thanks for this!

**Pull request merged**